Documentation: add guide for pysimCoder integration with NuttX (#4319)
apache/incubator-nuttx: Documentation/guides/pysimcoder.rst

.. include:: /substitutions.rst
.. _pysimcoder:

pysimCoder integration with NuttX
=================================

`PysimCoder <https://github.com/robertobucher/pysimCoder>`_ is as an open source

> Rapid Control Application Development Tool which is abe to transfer block diagrams into C code.

Reviewer: ```suggestion
Rapid Control Application Development Tool which is able to transfer block diagrams into C code.
```

Combined with NuttX, it can be used in real time control application and as an alternative to
expensive licensed programs and prototyping platforms. Example of `DC motor control application

> <https://www.youtube.com/watch?v=6HlGk3ecPNQ>`_ with PID controller and block blocks for encoder,

Reviewer: double word "block(s)"?

PWM, GPIO and sending data over TCP to real time plotter can be seen on `NuttX Channel 
<https://www.youtube.com/channel/UC0QciIlcUnjJkL5yJJBmluw>`_.

This documentation describes the steps that are needed to run application generated by pysimCoder
on NuttX and also keeps the track of peripherals that are supported by pysimCoder for NuttX RTOS.

Peripheral Support
------------------

The following list shows the peripherals and fuctionalities supported in pysimCoder for NuttX RTOS.

==========  =======================
Peripheral  Notes
==========  =======================
ADC
CAN         Including SocketCAN
DAC
ENC
GPIO
PWM         Multichannel support
UART        Serial output
Sensors     Basic support for DHTXX
TCP
UDP
==========  =======================

> Please note that the actual support for NuttX peripherals can be wider that it is mentioned here

Reviewer: ```suggestion
Please note that the actual support for NuttX peripherals can be wider than what is mentioned here
```

in case this documentation was not updated when new fuctionalities were added to pysimCoder.

NuttX Configuration
-------------------

Several configuration options are neccessary to be set in order to succesfully compile pysimCoder

> with NuttX. The list is following:

Reviewer: ```suggestion
with NuttX. The list is the following:
```
